### PR TITLE
Add the ed25519 curve

### PR DESCRIPTION
**Pull request by weikengchen**
## Description

Needs a decision from Pratyush on whether or not we want to add ed25519.

In addition, related to #115, if we add this curve, we probably should add the curve25519 curve soon, which is to avoid users from thinking ed25519 is curve25519's twisted Edwards curve.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (master)
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests
- [x] Updated relevant documentation in the code
- [x] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by weikengchen** [comment]
@arkworks-rs/algebra This one is ready for review.

File: ed25519/src/curves/mod.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+impl TECurveConfig for EdwardsParameters {
+    /// COEFF_A = -1
+    const COEFF_A: Fq = MontFp!("-1");
+
+    /// COEFF_D = -121665 / 121666
+    const COEFF_D: Fq =
+        MontFp!("37095705934669439343138083508754565189542113879843219016388785533085940283555");
+
+    /// Generated randomly
+    const GENERATOR: EdwardsAffine = EdwardsAffine::new_unchecked(GENERATOR_X, GENERATOR_Y);
```
Comment (Pratyush) [inline ed25519/src/curves/mod.rs:39]: Was this generated randomly?

File: ed25519/src/curves/mod.rs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+impl TECurveConfig for EdwardsParameters {
+    /// COEFF_A = -1
+    const COEFF_A: Fq = MontFp!("-1");
+
+    /// COEFF_D = -121665 / 121666
+    const COEFF_D: Fq =
+        MontFp!("37095705934669439343138083508754565189542113879843219016388785533085940283555");
+
+    /// Generated randomly
```
Comment (weikengchen) [inline ed25519/src/curves/mod.rs:38]:
```suggestion
    /// Standard generators from https://neuromancer.sk/std/other/Ed25519.
```

**Comment by weikengchen** [comment]
fixed the comments:
```
    /// Standard generators from https://neuromancer.sk/std/other/Ed25519.
```

**Review by Pratyush — APPROVED** [review]

File: ed25519/src/curves/mod.rs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+impl TECurveConfig for EdwardsParameters {
+    /// COEFF_A = -1
+    const COEFF_A: Fq = MontFp!("-1");
+
+    /// COEFF_D = -121665 / 121666
+    const COEFF_D: Fq =
+        MontFp!("37095705934669439343138083508754565189542113879843219016388785533085940283555");
+
+    /// Standard generators from https://neuromancer.sk/std/other/Ed25519.
```
Comment (weikengchen) [inline ed25519/src/curves/mod.rs:38]:
```suggestion
    /// Standard generators from <https://neuromancer.sk/std/other/Ed25519>.
```

**Pull request merged**